web/src/ui_init.js

```diff
@@ -787,6 +787,11 @@ $(async () => {
                 data,
                 success: resolve,
                 error(xhr) {
+                    $("#app-loading-middle-content").hide();
+                    $("#app-loading-bottom-content").hide();
+                    $(".app").hide();
+                    $("#app-loading-error").css({visibility: "visible"});
+
                     blueslip.error("Spectator failed to register", {
                         status: xhr.status,
                         body: xhr.responseText,
```

Comment:
We can probably drop this from being an `error` in this case, since we've somewhat handled it in the UI.  Ideally, we would also rewrite to not use `await`, so that the `reject()` doesn't also turn into an uncaught exception.

Reply:
Dropped the `error` and removed the `await`.